### ISSUE TITLE
Ac.Composite Error Handling Hides Errors from Parent

### ISSUE THREAD

**Issue by ooskapenaar** [YahooArchive/mojito#865]
In v0.5.0  composite.common.js we see the following code, which effectively hides any child errors from the caller. 

If the parent is to be allowed to "fail-fast", which I think is a good thing, then mojito must pass errors up to the parent and not swallow them like this. 

Swallowing of errors by mojito should only be allowed at the top level (i.e. to clean up after a messy parent).

``` javascript
       error: function(err) {
            Y.log("Error executing child mojit at '" + this.id + "':", 'error',
                    NAME);
            if (err.message) {
                Y.log(err.message, 'error', NAME);
            } else {
                Y.log(err, 'error', NAME);
            }
            if (err.stack) {
                Y.log(err.stack, 'error', NAME);
            }
            // Pass back some empty data so we don't fail the composite
            this.done('');
```

I am going to patch this code, by passing an error object back as the data, then the parent can sort out the mess, an alternative would be to throw an error and disrupt the composite process completely:

``` javascript

           if( err instanceof Error ) {
                this.done( err );
            } else {
                this.done( new Error( err ) );
            }

```

Opinions, ideas?

**Comment by caridy** [comment]
Adding this to trello, tentative for 0.5.4:

https://trello.com/card/issue-gh865-composite-ac-composite-error-handling-hides-errors-from-parent/504e5e3b7cd1578456fc430f/367

**Comment by rwaldura** [comment]
see also issue #25

**Comment by caridy** [comment]
will be available in mojito@0.5.5 scheduled for next week.
